.circleci/config.yml

```diff
@@ -0,0 +1,121 @@
+version: 2.1
+
+scala_212: &scala_212
+  SCALA_VERSION: 2.12.8
+
+scala_213: &scala_213
+  SCALA_VERSION: 2.13.0
+
+jdk_8: &jdk_8
+  JDK_VERSION: 8
+
+jdk_11: &jdk_11
+  JDK_VERSION: 11
+
+machine_ubuntu: &machine_ubuntu
+  machine:
+    image: ubuntu-1604:201903-01
+
+install_jdk: &install_jdk
+  - run:
+      name: Install JDK
+      command: |
+        while $(ps aux | grep -i ' apt ' | grep -v grep > /dev/null); do sleep 1; done # Wait for apt to be ready
+        sudo add-apt-repository ppa:openjdk-r/ppa -y
+        sudo apt update
+        sudo apt install openjdk-${JDK_VERSION}-jdk -y
+
+load_cache: &load_cache
+  - restore_cache:
+      key: sbt-nio-cache
+
+save_cache: &save_cache
+  - save_cache:
+      key: sbt-nio-cache
+      paths:
+        - "~/.ivy2/cache"
+        - "~/.sbt"
+        - "~/.m2"
+
+lint: &lint
+  steps:
+    - checkout
+    - <<: *load_cache
+    - <<: *install_jdk
+    - run:
+        name: Lint code
+        command: ./sbt ++${SCALA_VERSION}! check
+    - <<: *save_cache
+
+test: &test
+  steps:
+    - checkout
+    - <<: *load_cache
+    - <<: *install_jdk
+    - run:
+        name: Run tests
+        command: |
+          docker-compose up -d
+          ./sbt ++${SCALA_VERSION}! test
+    - <<: *save_cache
+
+jobs:
+  lint:
+    <<: *lint
+    <<: *machine_ubuntu
+    environment:
+      - <<: *scala_212
+      - <<: *jdk_8
+
+  test_212_jdk8:
+    <<: *test
+    <<: *machine_ubuntu
+    environment:
+      - <<: *scala_212
+      - <<: *jdk_8
+
+  test_212_jdk11:
+    <<: *test
+    <<: *machine_ubuntu
+    environment:
+      - <<: *scala_212
+      - <<: *jdk_11
+
+  test_213_jdk8:
+    <<: *test
+    <<: *machine_ubuntu
+    environment:
+      - <<: *scala_213
+      - <<: *jdk_8
+
+  test_213_jdk11:
+    <<: *test
+    <<: *machine_ubuntu
+    environment:
+      - <<: *scala_213
+      - <<: *jdk_11
+
+workflows:
+  version: 2
+  build:
+    jobs:
+      - lint:
+          filters:
+            tags:
+              only: /^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/
```

Comment:
You can remove this duplication via anchors (and we should make an effort to do that across the org @softinio). Checkout this [example](https://github.com/zio/zio/blob/master/.circleci/config.yml#L96).